service/history/engine/engineimpl/historyEngine.go

```diff
@@ -787,7 +787,17 @@ func (e *historyEngineImpl) startWorkflowHelper(
 		createMode,
 		prevRunID,
 		prevLastWriteVersion,
+		persistence.CreateWorkflowRequestModeNew,
 	)
+	if t, ok := err.(*persistence.DuplicateRequestError); ok {
```

Comment:
nit: prefer `errors.Is(err, persistence.DuplicateRequestError)`

Reply:
need type assertion, because we need to retrieve runID from the error

Reply:
`errors.As` would do the job then

Reply:
yes, my apologies, `errors.As`, though be careful to test it, it's double-pointer API is kinda confusing. 
```
var e *persistence.DuplicateRequestError
if errors.As(err, &e) {
  ...
}
```